pkg/sriov/sriov.go

```diff
@@ -245,8 +321,66 @@ func (s *sriovManager) ApplyVFConfig(conf *sriovtypes.NetConf) error {
 	}
 
 	// 2. Set mac address
+	if conf.MAC != "" {
+		hwaddr, err := net.ParseMAC(conf.MAC)
+		if err != nil {
+			return fmt.Errorf("failed to parse MAC address %s: %v", conf.MAC, err)
+		}
+
+		// Save the original MAC address before overriding it
+		vf := getVfInfo(pfLink, conf.VFID)
+		if vf == nil {
+			return fmt.Errorf("failed to find vf %d", conf.VFID)
+		}
+		conf.OriginalMAC = vf.Mac.String()
+
+		if err = s.nLink.LinkSetVfHardwareAddr(pfLink, conf.VFID, hwaddr); err != nil {
+			return fmt.Errorf("failed to set MAC address to %s: %v", hwaddr, err)
+		}
+	}
+
+	// 3. Set link rate.
+	if conf.MaxTxRate != nil {
+		if err = s.nLink.LinkSetVfTxRate(pfLink, conf.VFID, *conf.MaxTxRate); err != nil {
+			return fmt.Errorf("failed to set vf %d max_tx_rate to %d Mbps: %v", conf.VFID, conf.MaxTxRate, err)
+		}
+	}
 
-	// 3. Set link rate
+	// 4. Set spoofchk flag
```

Comment:
it seem you can do it in a loop gove ove all the attributes in conf and call the right netlink command